gradio/event_queue.py

```diff
@@ -195,9 +195,13 @@ async def send_estimation(
             rank:
         """
         estimation.rank = rank
-        estimation.rank_eta = round(
-            (estimation.rank + 1) * cls.AVG_CONCURRENT_PROCESS_TIME + cls.AVG_PROCESS_TIME
-        )
+
+        if cls.AVG_CONCURRENT_PROCESS_TIME is not None:
+            estimation.rank_eta = (
+                estimation.rank * cls.AVG_CONCURRENT_PROCESS_TIME + cls.AVG_PROCESS_TIME
+            )
+            if None not in cls.ACTIVE_JOBS:
+                estimation.rank_eta += cls.AVG_CONCURRENT_PROCESS_TIME
```

Comment:
Good perspective, a comment would be nice though, had hard time to understand what it does.
```suggestion
                # Add estimated amount of time for a thread to get empty 
                estimation.rank_eta += cls.AVG_CONCURRENT_PROCESS_TIME 
```

Reply:
Makes sense